e2e-tests/src/test/emergency_finalizer.rs

```diff
@@ -24,3 +34,117 @@ async fn set_emergency_finalizer_test() -> anyhow::Result<()> {
     assert_eq!(current_finalizer, Some(public));
     Ok(())
 }
+
+async fn finalize_from_to<C: AsConnection + Sync>(
+    connection: &C,
+    from: BlockHash,
+    to: BlockHash,
+    finalizer: AlephKeyPair,
+) -> anyhow::Result<()> {
+    let from = connection.get_block_number(from).await?.unwrap();
+    let to = connection.get_block_number(to).await?.unwrap();
+
+    for i in (from + 1)..=to {
+        let hash = connection.get_block_hash(i).await?.unwrap();
+        connection.emergency_finalize(i, hash, finalizer).await?
+    }
+    Ok(())
+}
+
+async fn setup() -> anyhow::Result<AlephKeyPair> {
+    let config = setup_test();
+    let (finalizer, _seed) = AlephKeyPair::generate();
+    let connection = config.create_root_connection().await;
+
+    connection
+        .set_emergency_finalizer(finalizer.public().into(), TxStatus::Finalized)
+        .await?;
+    connection
+        .wait_for_n_sessions(2, BlockStatus::Finalized)
+        .await;
+
+    let accounts = get_validators_keys(config);
+
+    let new_validators: Vec<AccountId> = accounts
+        .iter()
+        .map(|pair| pair.signer().public().into())
+        .collect();
+
+    let seats = CommitteeSeats {
+        reserved_seats: 4,
+        non_reserved_seats: 0,
+        non_reserved_finality_seats: 0,
+    };
+
+    connection
+        .change_validators(
+            Some(new_validators[0..4].to_vec()),
+            Some(vec![]),
+            Some(seats.clone()),
+            TxStatus::InBlock,
+        )
+        .await?;
+    connection.wait_for_n_eras(1, BlockStatus::Finalized).await;
+    connection
+        .change_validators(
+            Some(new_validators[2..].to_vec()),
+            Some(vec![]),
+            Some(seats),
+            TxStatus::InBlock,
+        )
+        .await?;
+    disable_validators(&[0, 1]).await?;
+
+    Ok(finalizer)
+}
+
+/// Tests emergency finalizer. Runs on 6 nodes 0-6.
+/// * Setup finalizer
+/// * setup 0-4 to be validators in the next era
+/// * setup 2-6 to be validators in the next next era
+/// * disable 0-1.
+/// * wait for next era
+/// * check if finalization stopped
+/// * use finalizer to advance into next-next era
+/// * check if finalization resumed
+#[tokio::test]
+async fn chain_dead_scenario() -> anyhow::Result<()> {
+    let config = setup_test();
+    let finalizer = setup().await?;
+    let connection = config.create_root_connection().await;
+
+    let last_finalized = connection.get_finalized_block_hash().await?;
+    let last_finalized_before = connection.get_block_number(last_finalized).await?.unwrap();
+    sleep(Duration::from_secs(40));
+    let mut last_finalized = connection.get_finalized_block_hash().await?;
+    let last_finalized_now = connection.get_block_number(last_finalized).await?.unwrap();
+
+    assert!(
+        last_finalized_now - last_finalized_before <= 20,
+        "at most 20 blocks can be created after finalization stops"
```

Comment:
That's a bit of a _non sequitur_ here, since you check finalized block, which, uh, shouldn't be finalized at all after finalization stops, right? I'm not sure what the condition here should be, but at least the justification doesn't make sense, so is likely to be confusing in the future.

Reply:
oh, I will change that last_finalized_now to us the latest best_block instead.

Reply:
Huh, should that actually work? I thought disabling validators only happened after a session or two (since it's a key change), and here we immediately expect it to have worked? Or is there a hidden (or just missed by me :P) wait somewhere there?

Reply:
there is a 2 session wait in disable function